Growing animation settings; new bouncy animation

va-0 opened the pull request:
I figured a new animation would be fun to add; this one incorporates acceleration, squash n' stretch, and wobble.

[This is the traditional growing-animation behavior (known as LINEAR)](https://i.imgur.com/W95co94.mp4)
[This is a new optional growing-animation behavior (known as BOUNCY)](https://i.imgur.com/VbfsYi3.mp4)

This is the setting to change between them:
![Setting to change between the feeding animations](https://i.imgur.com/If8Brsn.png)

It also seemed useful to have options for growth/degrowth in the stress-tester
![Stress test button changes](https://i.imgur.com/dFM4cK2m.png)

Does everything look alright?

Poobslag commented:
Thanks so much for your contribution!

The options for growth/degrowth already can be tested in the RestaurantPuzzleSceneDemo.tscn and RestaurantViewDemo.tscn. The controls are described at the top of the script files.

```
## Keys:
## 	[D]: Ring the doorbell
## 	[F]: Feed the creature
## 	[I]: Launch an idle animation
## 	[V]: Say something
## 	[N]: Change the nametag names
## 	[1-9,0]: Change the creature's size from 10% to 100%
```

I agree it's useful to have this feature in the Stress Tester now -- but I'd like to keep the stress tester simple. Honestly I'd like to delete it entirely -- but I don't have a better way of testing performance issues on my android phone.

Other than that I love the change and I'd probably make a few other tweaks

  * Change the default creature growth mode from LINEAR to BOUNCY (yours looks better, and in a few months if nobody prefers the older one, I'd say we can just get rid of the LINEAR setting altogether)
  * Change `lerp_speed` from 0.01 to 0.02. In the puzzles, creatures grow VERY incrementally, like 1-2 pixels for every snack they eat. It's strange to have the creature undulating for 2 seconds when they only grew in size by 1 or 2 pixels, so having this animation on the snappier side fits better.
  * A nitpicky thing, but the SettingsMenu connections all got rearranged which makes it harder to trace later when connections are added or removed. ...This is a side effect of Godot #52493 (https://github.com/godotengine/godot/issues/35084) which has allegedly been fixed a few months ago. If that's true, then maybe I can sort these connections en masse and stop alphabetizing them myself with a lame little script I run (ugh) but in the meantime I'd like to leave the connections alone

I don't know what the appropriate Godot etiquette is for editing pull requests, and I'm happy to make these tweaks myself before merging it in. Do I clone this branch into my repository and make the appropriate changes, or what is the typical process?

Thanks so much for your contribution, this looks so cool!

va-0 commented:
It wasn't too difficult, so I went and did those tweaks quickly (though I changed lerp_speed to 0.05 because 0.02 did not make much difference).
I believe when creating this PR I checked the "Allow maintainers to edit this pull request" option, [so I presume you making changes would be related to this](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
Hope that's all good :)

Poobslag commented:
Thanks -- sounds good! I followed your link but evidently editing a GitHub pull request requires a lot of weird thorny steps I've never done before

- [x] Install the GitHub CLI (done)
- [ ] Generate and secure a GitHub API authentication token somewhere on my filesystem
- [ ] Authenticate against github.com by reading the token from a file
- [ ] Utilize the GitHub CLI to check out the PR (`gh pr checkout`)?
- [ ] Utilize the GitHub CLI to edit the PR (I have no clue how to do this, but presumably there is equivalent of the `git commit` and `git push` commands?)
- [ ] Squash the branch (presumably some github CLI equivalent of `git rebase`?)
- [ ] Merge it into my repo (i can probably do this over the web)

I'm well-versed in the Git command line, but never messed with the GitHub CLI so this is all new to me! I will try to do some of it next week or possibly in mid-January -- but i have some other development tasks. Sorry this ended up being so difficult!

va-0 commented:
Gotcha, that's all right!
Also, I swapped [\[snappier bounce, changed default, cleaned code*\]](https://github.com/Poobslag/turbofat/pull/1888/commits/ac29ec7c5c942f4dd8a8e7160702c72e3fab4d04) around because it was senselessly modifying the stress test scene (I thought re-basing would clean up the commits on that, but force-pushing still dirties it up on github, oh well ;) )

Poobslag commented on project/src/main/world/creature/creature-visuals.gd:
```diff
@@ -70,6 +70,14 @@ var comfort := 0.0 setget set_comfort
+
+## Stores the normal scale for the creature (right now just 0.6 for squirrels)
+## Changed upon creature_visuals.rescale(), used so we can squash and stretch based on this value
+var _base_scale: Vector2 = Vector2(1.0, 1.0)
```
Use the `Vector2.ONE` constant instead of hard-coding `Vector2(1.0, 1.0)`

Poobslag commented on project/src/main/world/creature/creature-visuals.gd:
```diff
@@ -70,6 +70,14 @@ var comfort := 0.0 setget set_comfort
 
+## For bouncy feeding animation, allowing inertia and acceleration.
+## (See: enum GraphicsSettings.FeedingAnimation and creature_visuals.update_fattening_animation)
+var _fattening_intertia: float = 0
```
Typo; the world is spelled `inertia`

Poobslag commented:
I've merged these changes via #1890 -- Presumably there is a better way to edit and merge branches from a different repository but after playing around for 30 minutes I couldn't figure it out.

Here's what I did instead:

```bash
git remote add va-0 git@github.com:va-0/turbofat.git
git pull
git checkout va-0/main
git checkout -b squash-and-stretch
git commit -v # committed my changes here
git rebase -i main # squashed your changes here
git push -f origin "$(git rev-parse --abbrev-ref HEAD)" # force push the squash to my branch
```

I kept your contribution in its own commit in the Turbo Fat codebase so you can see it 😀 thanks so much! Let me know if there's an easy way to do all this stuff since I don't work with multiple repositories very often.

https://github.com/Poobslag/turbofat/commit/8d984b977d3b406dbf398425cac64ada6969439d

Poobslag commented:
Subsumed via #1890
